infra/modules/database/dms-instance.tf

```diff
@@ -1,6 +1,22 @@
 # DMS replication instance and endpoint connections
 
-resource "aws_dms_replication_instance" "simpler_db" {
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets
+data "aws_subnets" "all" {
+  filter {
+    name   = "vpc-id"
+    values = [var.vpc_id]
+  }
+}
+
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_subnet_group
+resource "aws_dms_replication_subnet_group" "subnet" {
+  replication_subnet_group_description = "${var.environment_name} replication subnet group"
+  replication_subnet_group_id          = "${var.environment_name}-replication-subnet-group"
+  subnet_ids                           = data.aws_subnets.all.ids
+}
```

Comment:
Not sure why so many AWS services need their own special purpose subnets

Also not sure if this should be attached to *all* the subnets or just the *private* ones.

Reply:
would it be better to add some of these services to preexisting subnet groups rather than creating their own? (e.g. all DMS and DB related resources in the same subnet groups)

Reply:
@aplybeah Yes but we're constrained by AWS's architecture here, which requires a specific DMS subnet. That's what this resource is for.

So this isn't anything we can change, more so it's a comment about the way AWS architects it's services. GCP does the opposite thing, where it almost always uses the standard VPC subnets that you provide.

Reply:
I need to consult the architecture docs for the template, but I wonder if the original plan was to have all the services in all subnets and then use security groups to selectively add access/permissions/etc